src/editor/Model/WindowManager.re

```diff
@@ -0,0 +1,48 @@
+open Revery_UI;
+open Oni_Core;
+
+type layout =
+  | VerticalLeft
+  | VerticalRight
+  | HorizontalTop
+  | HorizontalBottom;
+
+module WindowSplitId =
+  Revery.UniqueId.Make({});
+
+type split('a) = {
```

Comment:
This approach is pretty much on the same page as what I was thinking 👍 Looks good!